src/main/java/com/frameworkium/core/ui/ExtraExpectedConditions.java

```diff
@@ -21,6 +21,39 @@
 public class ExtraExpectedConditions {
 
     /**
+     * An expectation for checking that N elements (from a list) are present on the web page that match the locator
+     * are visible. Visibility means that the elements are not only displayed but also have a height
+     * and width that is greater than 0.
+     *
+     * @param elementsToCheck how many elements in a <code>List&lt;WebElement&gt;</code> to check
+     * @param elements list of WebElements
+     * @return the list of WebElements once they are located
+     */
+    public static ExpectedCondition<List<WebElement>> visibilityOfNElements(
+            final List<WebElement> elements,int elementsToCheck) {
+
+        int finalElementsToCheck = elementsToCheck == 0 ? elements.size() : elementsToCheck;
+
+        return new ExpectedCondition<List<WebElement>>() {
```

Comment:
You can use the `expectedCondition` method at the bottom of this class to reduce the boilerplate code here.